crates/noirc_errors/src/debug_info.rs

```diff
@@ -1,39 +1,41 @@
+use acvm::acir::circuit::OpcodeLocation;
+use acvm::compiler::AcirTransformationMap;
+
+use serde_with::serde_as;
+use serde_with::DisplayFromStr;
 use std::collections::BTreeMap;
 
 use crate::Location;
 use serde::{Deserialize, Serialize};
 
+#[serde_as]
 #[derive(Default, Debug, Clone, Deserialize, Serialize)]
 pub struct DebugInfo {
     /// Map opcode index of an ACIR circuit into the source code location
-    pub locations: BTreeMap<usize, Vec<Location>>,
+    #[serde_as(as = "BTreeMap<DisplayFromStr, _>")]
+    pub locations: BTreeMap<OpcodeLocation, Vec<Location>>,
 }
 
 impl DebugInfo {
-    pub fn new(locations: BTreeMap<usize, Vec<Location>>) -> Self {
+    pub fn new(locations: BTreeMap<OpcodeLocation, Vec<Location>>) -> Self {
         DebugInfo { locations }
     }
 
     /// Updates the locations map when the circuit is modified
-    ///
-    /// When the circuit is generated, the indices are 0,1,..,n
```

Comment:
can you add a doc comment explaining what this method is used for, similar to the one that was removed?